consumer/src/test/kotlin/streams/StreamsSinkConfigurationTest.kt

```diff
@@ -1,14 +1,79 @@
 package streams
 
+import org.junit.Before
 import org.junit.Test
-import org.neo4j.configuration.Config
+import org.mockito.Mockito
+import org.neo4j.dbms.api.DatabaseManagementService
+import org.neo4j.graphdb.GraphDatabaseService
+import org.neo4j.graphdb.ResourceIterator
+import org.neo4j.graphdb.Result
+import org.neo4j.graphdb.Transaction
+import org.neo4j.logging.NullLog
+import streams.config.StreamsConfig
 import streams.service.TopicType
+import streams.service.TopicValidationException
 import kotlin.test.assertEquals
 import kotlin.test.assertFalse
-import kotlin.test.assertTrue
 
 class StreamsSinkConfigurationTest {
 
+    lateinit var streamsConfig: StreamsConfig
+    val defalutDbName = "neo4j"
+
+    @Before
+    fun setUp() {
+        val dbms = Mockito.mock(DatabaseManagementService::class.java)
+        val db = Mockito.mock(GraphDatabaseService::class.java)
+        val tx = Mockito.mock(Transaction::class.java)
+        val result = Mockito.mock(Result::class.java)
+        val resource = Mockito.mock(ResourceIterator::class.java) as ResourceIterator<String>
+        Mockito.`when`(dbms.database(Mockito.anyString())).thenReturn(db)
+        Mockito.`when`(db.beginTx()).thenReturn(tx)
+        Mockito.`when`(tx.execute(Mockito.anyString())).thenReturn(result)
+        Mockito.`when`(result.columnAs<String>(Mockito.anyString())).thenReturn(resource)
+        Mockito.`when`(resource.hasNext()).thenReturn(true)
+        Mockito.`when`(resource.next()).thenReturn(defalutDbName)
+        streamsConfig = StreamsConfig(NullLog.getInstance(), dbms)
+    }
+
+    @Test
+    fun `should manage only topics for default db`() {
+        val pollingInterval = "10"
+        val topicKey = "streams.sink.topic.cypher.myTopic"
+        val topicValue = "MERGE (n:Label{ id: event.id })"
+        val topicKeyNeo = "streams.sink.topic.cypher.myTopicNeo.to.neo4j"
+        val topicValueNeo = "MERGE (n:Neo4j{ id: event.id })"
+        val topicKeyFoo = "streams.sink.topic.cypher.myTopicFoo.to.foo"
+        val topicValueFoo = "MERGE (n:Foo{ id: event.id })"
+        val config = mapOf("streams.sink.polling.interval" to pollingInterval,
+                topicKey to topicValue,
+                topicKeyNeo to topicValueNeo,
+                topicKeyFoo to topicValueFoo)
+        streamsConfig.config.putAll(config)
+        val streamsSinkConf = StreamsSinkConfiguration.from(streamsConfig, defalutDbName)
+        val cypherTopics = streamsSinkConf.topics.asMap()[TopicType.CYPHER] as Map<String, String>
+        assertEquals(mapOf("myTopic" to topicValue, "myTopicNeo" to topicValueNeo), cypherTopics)
+    }
+
```

Comment:
Add test -- attempt to do this should fail:

streams.sink.topic.cypher.whatever.to.system=whatever

Should always fail.   "system" is not a valid sink target.

Reply:
We block the creation of the sink/source plugins at `Extension` level:

https://github.com/neo4j-contrib/neo4j-streams/blob/e17d6d64d33a8d1ae61d8a07f6a6193a06f2c551/producer/src/main/kotlin/streams/StreamsExtensionFactory.kt#L50-L52

and

https://github.com/neo4j-contrib/neo4j-streams/blob/e17d6d64d33a8d1ae61d8a07f6a6193a06f2c551/consumer/src/main/kotlin/streams/StreamsEventSinkExtensionFactory.kt#L45-L47

Do you want to add a check also here?

Reply:
Sorry if I missed it, if the check is already in there it isn't double-needed